### ISSUE TITLE
automatic backup of open maps

### ISSUE THREAD

**Issue by daveiro** [phase1geo/Minder#446]
Me and, it seems, several other people, sometimes get random crashes while working in maps and can lose minutes or hours of work. The ones I've had are difficult to reproduce. I read problems with valac might cause crashes like these. In any case, I would suggest a feature/enhancement is added to Minder (which exists and has saved me a few times in FreeMind): the option to automatically create backup of open files every x (to be defined in some general config option) minutes. The backup file name could be something like originalname_202101131948 (YYYYMMDDHHMM) for example. And it could keep the backup files for y (to be defined in some general config option) days or indefinitely.

**Comment by daveiro** [comment]
PS: I just noticed this feature is offered in the latest master version, which is great and has just saved me now in a strange "tree lost" case which I'll try to reproduce and file a bug... but I would suggest these backups are not stored on the same folder as the original file but in an "internal folder" like (in case of linux): .local/share/minder/backup

**Comment by daveiro** [comment]
PS2: I now noticed that this is not "automatic regular backup" but some backup that seems to have been generated because minder notified me a certain file was changed and requested me to either reload or overwrite, which is strange as I only had one instance of the file open... reloading did not work at first so then I selected overwrite (or the other way around, don't remember well now) and I suppose this generated the backups; so my request for "regular automatic backups" is still on, thanks!

**Comment by harcesz** [comment]
this would be useful; a crash on cut and paste just removed a lot of my work, old file got overwritten by one without a huge chunk of what I had and I've no way of going back

**Comment by daveiro** [comment]
Indeed, it seems several people are loosing a lot of their work because of such cut-and-paste crashes. Happens to me randomly and occasionally.

And this problem is also reported in [this issue](https://github.com/phase1geo/Minder/issues/482), where several users have requested an option to disable auto-save, which is one of the culprits of lost work by crashes.

If auto-save was off, at least the last saved version would have the content.

Nevertheless, I would say that automatic backup is an essential option too, especially for users who prefer to keep auto-save on and while this situation of random crashes is not solved.
